pytorch_lightning/overrides/distributed.py

```diff
@@ -123,12 +124,31 @@ class IndexBatchSamplerWrapper:
     """This class is used to wrap a :class:`torch.utils.data.BatchSampler` and capture its indices."""
 
     def __init__(self, sampler: BatchSampler) -> None:
+        self.seen_batch_indices: List[List[int]] = []
         self._sampler = sampler
-        self.batch_indices: Optional[List[int]] = None
+        self._batch_indices: List[int] = []
+
+    @property
+    def batch_indices(self) -> List[int]:
```

Comment:
reviewer: the bugfix requires a slight redesign here. This public attribute is kept backward compatible but is not useful anymore, hence the deprecation.

Reply:
If this is going into the bugfix release, it should mention the bugfix version it was ddprecated on, not 1.6

Reply:
right, adjusted the range to v1.5.5 - v1.7